big-o/problem-set-big-o.md

```diff
@@ -5,3 +5,549 @@
+
+In Big O terms, the `print` statement will get executed _(17 * n)_ number of times, where _n_ is the length of `input_list`. While determining time complexity, we drop the constants. So, the time complexity of this algorithm is _O(n)_, where n is the length of `input_list`. In other words, the time complexity is linearly proportional to the input size.
+
+##### !end-explanation
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 8 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: multiple-choice
+* id: 5qCZ8q
+* title: Big O
+##### !question
+
+What is the time complexity of the following piece of code?
+
+```python
+def greet_friends(input_list):
+    count = len(input_list)
+    i = 0
+    while i < count:
+        for j in range(count):
+            print(f"Hello, Friend #{i+1} in {j+1}!")
+        i += 1
+```
+
+##### !end-question
+##### !options
+
+* O(1)
+* O(log n)
+* O(n)
+* O(nlog n)
+* O(n^2)
+* O(n^3)
+
+##### !end-options
+##### !answer
+
+* O(n^2)
+
+##### !end-answer
+##### !explanation
+
+There are two nested loops in this method. Each loop runs _n_ number of times where _n_ is the length of `input_list`. Since the loops are nested, the loop will run _n * n_ number of times.
+
+So, the times complexity of this algorithm is _O(n ^ 2)_ where _n_ is the length of the input test_array. In other words, the time complexity is _quadratic_, which is a _polynomial_ time complexity.
+
+##### !end-explanation
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 9 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: multiple-choice
+* id: rkV75c
+* title: Big O
+##### !question
+
+What is the time complexity of the following piece of code?
+
+```python
+def greet_friends(input_list):
+    count = len(input_list)
+    k = 0
+    while k != count:
+        i = 0
+        while i < count:
+            for j in range(count):
+                print(f"Hello, Friend #{i+1} in #{j+1}!")
+            i += 1
+        k += 1
+```
+
+##### !end-question
+##### !options
+
+* O(1)
+* O(log n)
+* O(n)
+* O(nlog n)
+* O(n^2)
+* O(n^3)
+
+##### !end-options
+##### !answer
+
+* O(n^3)
+
+##### !end-answer
+##### !explanation
+
+There are three nested loops in this method, each running _n_ times where _n_ is the length of `input_list`. The time complexity will therefore be _n * n * n_ or in Big O terms _O(n ^ 3)_. In other words, the time complexity will be _cubic_, which is a _polynomial_ time complexity.
+
+##### !end-explanation
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 10 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: multiple-choice
+* id: b9HqKs
+* title: Big O
+##### !question
+
+What is the time complexity of the following piece of code?
```

Comment:
Consider explicitly stating that we can assume that the input list is in sorted order.

```suggestion
What is the time complexity of the following piece of code? Assume that `input_list` is in sorted order.
```